src/sync/sync_user.cpp

```diff
@@ -59,14 +59,14 @@ RealmJWT::RealmJWT(const std::string& token)
 {
     auto parts = split_token(token);
 
-    auto json = std::get<bson::BsonDocument>(bson::parse(base64_decode(parts[1])));
+    auto json = (bson::BsonDocument)bson::parse(base64_decode(parts[1]));
```

Comment:
Using this cast syntax produces warnings when building with `-Wold-style-casts`, which when building with CocoaPods can end up trickling down to our source files. `static_cast<T>()` avoids that (and doesn't do some dangerous conversions which C casts can do).